main/static/script.js

```diff
@@ -16,20 +19,47 @@ module.controller("mainCtrl", function($scope, $http) {
 		    $scope.queryResult = data;
 		  });
 	};
+
 	$scope.$watch("queryResult", resultUpdate);
+
+	function readHash() {
+		var urlQuery = window.location.hash
+		if (urlQuery != "") {
+			// Drop the leading '#'
+			urlQuery = urlQuery.substring(1);
+			// The remainder is the query to perform.
+			// Decode it (if neccesarry).
+			$scope.queryText = decodeURIComponent(urlQuery);
+			$scope.onSubmitQuery();
+		}
+	}
+
+	readHash(); // Read the current hash (if present).
+
+	// Whenever the user changes the hash (such as pasting in a new URL), call readHash() again.
+	window.addEventListener("hashchange", readHash, false);
```

Comment:
Also, the same holds true even if the user types it in themselves, at least if the hash is already present in the URL.